Adding Contributing Guidelines (#21)
whitep4nth3r/pantherworld: CONTRIBUTING.md

# Contributing Guidelines

*Pull requests, bug reports, and feature requests are welcomed and highly encouraged!*

### Contents

- [General Guidelines](#-general-guidelines)
- [Questions](#-questions)
- [Opening an Issue](#-opening-an-issue)
- [Feature Requests](#-feature-requests)
- [Submitting Pull Requests](#-submitting-pull-requests)

## 📖 General Guidelines

For everything from pull requests, bug reports, features requests and code reviews the following applies:

**Be respectful of each other!** Abusive language or harassment is not welcome here!

## 💡 Questions

> The [The Claw Discord](https://discord.gg/theclaw) or [whitep4nth3r's twitch chat](https://www.twitch.tv/whitep4nth3r) are the places to be for any questions regarding the code or p4nth3rworld itself. 

Reviewer: ```suggestion
If you have any questions regarding the code in this project or the p4nth3rworld game itself, ask in [The Claw Discord](https://discord.gg/theclaw) or [whitep4nth3r's twitch chat](https://www.twitch.tv/whitep4nth3r).
```

> The [The Claw Discord](https://discord.gg/theclaw) or [Whitep4nth3r's twitch chat](https://www.twitch.tv/whitep4nth3r) are the places to be for any questions regarding the code or p4nth3rworld itself. 

Reviewer: ```suggestion
The [The Claw Discord](https://discord.gg/theclaw) or [whitep4nth3r's twitch chat](https://www.twitch.tv/whitep4nth3r) are the places to be for any questions regarding the code or p4nth3rworld itself. 
```

> Please don't open issues for questions about how to play the game.

Reviewer: ```suggestion
Please do not open issues for questions about how to play the game.
```

> Please don't open issues for questions to not create any clutter.

Reviewer: ```suggestion
Please don't open issues for questions about how to play the game.
```

## 📥 Opening an Issue

> If you find any issues with the p4nth3rworld site or p4nth3rworld the game we would love a well-written report!

Reviewer: ```suggestion
If you find any issues with the [p4nth3rworld front end app](https://p4nth3r.world) or p4nth3rworld the game, we would love a well-written report!
```

- **Do not open a duplicate issue!** Search through existing issues to see if your issue has previously been reported. If your issue exists, comment with any additional information you have. You may simply note "I have this problem too", which helps prioritize the most common problems and requests. 

> - **Prefer the use of [reactions](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)**, not comments, if you simply want to "+1" an existing issue.

Reviewer: ```suggestion
- **The use of [reactions](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** instead of comments is preferred, if you simply want to "+1" an existing issue.
```

- **Provide as much information as you can.** We don't have a  bug report template, just be thorough with your report and if possible add some images. It's also very usefull to add your twitch username if it differs from your github one!

## 💌 Feature Requests

We very much welcome feature requests! But do keep in mind that not every feature can or should be added. Also we very much encourage you to create a pull request yourself! We have only a few people working on this project and would love the help!

- **Do not open a duplicate feature request.** Search for existing feature requests first. If you find your feature (or one very similar) previously requested, comment on that issue.

## 🔁 Submitting Pull Requests

We very much welcome pull requests, and as previously mentioned, very much encourage people to create pull requests for features or issues they encountered themselves!

> - **Make isolated changes** When making a pull request make sure that all the changes within it are solely about this one feature or bug fix. Don't include multiple issues or features

Reviewer: ```suggestion
- **Make isolated changes** When making a pull request make sure that all the changes within it are solely about this one feature or bug fix. Do not include multiple issues or features. 

Try to follow existing patterns in the existing code when submitting a pull request. PRs completely refactoring the whole code style of the project will not be accepted.
```

- **The use of [reactions](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** instead of comments is preferred, if you simply want to "+1" an existing issue.

- **Provide as much information as you can.** We don't have a  bug report template, just be thorough with your report and if possible add some images. It's also very usefull to add your twitch username if it differs from your github one!

## 💌 Feature Requests

We very much welcome feature requests! But do keep in mind that not every feature can or should be added. Also we very much encourage you to create a pull request yourself! We have only a few people working on this project and would love the help!

- **Do not open a duplicate feature request.** Search for existing feature requests first. If you find your feature (or one very similar) previously requested, comment on that issue.

## 🔁 Submitting Pull Requests

We very much welcome pull requests, and as previously mentioned, very much encourage people to create pull requests for features or issues they encountered themselves!

> - **Make isolated changes** When making a pull request make sure that all the changes within it are solely about this one feature or bug fix. Do not include multiple issues or features. 

Reviewer: ```suggestion
- **Make isolated changes** .When making a pull request make sure that all the changes within it are solely about this one feature or bug fix. Do not include multiple issues or features. 
```
